lib/web_apis/web_apis.es6.js

```diff
@@ -0,0 +1,347 @@
+          typeName: 'String',
+        },
+        {
+          name: 'apiCatalog',
+          documentation: `The web catalog of this version of
+            browser, tested on the given operating system and version.
+            The json object should be the JSON object returned from
+            com.web.catalog.apiExtractor`,
+          typeName: 'JSON',
+        },
+      ],
+      code: function(browserName, browserVersion,
+        OSName, OSVersion, apiCatalog) {
+        let browser = this.Browser.create({
+          browserName: browserName,
+          browserVersion: browserVersion,
+          OS: OSName,
+          OSVersion: OSVersion,
+        });
+        let browserId = browser.id;
+        this.browsers.put(browser);
+        let interfaceNames = Object.keys(apiCatalog);
+        for (let i = 0; i < interfaceNames.length; i += 1) {
+          let interfaceName = interfaceNames[i];
+          for (let j = 0; j < apiCatalog[interfaceName].length; j += 1) {
+            let APIName = apiCatalog[interfaceName][j];
+            let webInterface = this.WebInterface.create({
+              interfaceName,
+              APIName,
+            });
+            let interfaceId = webInterface.id;
+            this.interfaces.put(webInterface);
+            this.browserAPIs.put(this.BrowserAPI.create({
+              browserId,
+              interfaceId,
+              id: `${browserId}.${interfaceId}`,
+            }));
+          }
+        }
+      },
+    },
+    {
+      name: 'getBrowserKeys',
+      documentation: `Asynch function to return an array of stored
+        browsers' key. browserName and version are optional arguments
+        to filter keys.`,
+      args: [
+        {
+          name: 'browserName',
+          documentation: `The optional argument that the returned browser
+            key will match the given name`,
+          typeName: 'String',
+        },
+        {
+          name: 'version',
+          documentation: `The optional argument that the returned browser
+            key will match the given version`,
+          typeName: 'String',
+        },
+      ],
+      returns: {
+        typeName: 'StringArray',
+        documentation: `Return a string array contains matching browser keys`,
+      },
+      code: function(browserName, version) {
+        return new Promise((resolve) => {
+          this.browsers.select().then((arrayDAO) => {
+            resolve(arrayDAO.a
+            .map((browser) => browser.getBrowserKey())
+            .filter((key) => {
+              if (browserName &&
+              key.split('_')[0].indexOf(browserName) !== 0) {
+                return false;
+              }
+              if (version &&
+              key.split('_')[1].indexOf(version) !== 0) {
+                return false;
+              }
+              return true;
+            }));
+          });
+        });
+      },
+    },
+    {
+      name: 'getBrowserId_',
```

Comment:
As above regarding need for this. We can query our dao for the `Browser` object and ask it for its key. Are you thinking that making such a query be synchronous is critical?